### ISSUE TITLE
No way to check if type is registered considering parents

### ISSUE THREAD

**Issue by samdark** [go-ozzo/ozzo-di#5]
Current implementation lacks a way to check if type is registed in either container or any of its parents while `build()` considers it.

**Comment by kolkov** [comment]
Good catch — `build()` internally checks parents for registration, but there's no public API to do the same check. Something like a `HasType(t reflect.Type) bool` method that walks the container chain would be straightforward to add.

I've taken over maintenance here. Is this still relevant for your use case?

**Comment by samdark** [comment]
I'm not personally using it right now. THe issue was found during creation of https://github.com/yiisoft/di
